### PR TITLE
Added New Shoe form/new shoe page

### PR DESCRIPTION
**Pull request by JusLafo**
- Added a new shoe page at /shoes/new
- Added to corresponding router endpoints

### CODE REVIEW

File: client/src/components/App.js [hunk 1]
```diff
@@ -28,7 +29,9 @@ const App = (props) => {
     <Router>
       <TopBar user={currentUser} />
       <Switch>
-        <Route exact path="/" component={ShoesListPage}/>
+        <Route exact path="/" component={ShoesListPage} />
+				<Route exact path="/shoes" component={ShoesListPage} />
+				<Route exact path="/shoes/new" component={NewShoeForm} />
```
Comment (nwalberts) [inline client/src/components/App.js]: this indentation is weird, lets fix it

File: client/src/components/NewShoeForm.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+				}
+			} else {
+				const body = await response.json()
+				console.log("Shoe created successfully.", body)
+				setShouldRedirect(true)
+			}
+		} catch (err) {
+			console.error(`Error in fetch: ${err.message}`)
+		}
+	}
```
Comment (nwalberts) [inline client/src/components/NewShoeForm.js]: please work with your group to edit your project's tab spacing

e.g. https://stackoverflow.com/questions/29972396/how-can-i-customize-the-tab-to-space-conversion-factor

this indentation is too large, we need to trim this back a smidge

File: client/src/components/NewShoeForm.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+					const newErrors = translateServerErrors(body.errors)
+					return setErrors(newErrors)
+				} else {
+					const errorMessage = `${response.status} (${response.statusText})`
+					const error = new Error(errorMessage)
+					throw error
+				}
+			} else {
+				const body = await response.json()
+				console.log("Shoe created successfully.", body)
```
Comment (nwalberts) [inline client/src/components/NewShoeForm.js]: lets remove this console.log

File: client/src/components/NewShoeForm.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+			[event.currentTarget.name]: event.currentTarget.value
+		})
+	}
+
+	const handleSubmit = event => {
+		event.preventDefault()
+		addNewShoe()
+	}
+
+	console.log(newShoe)
```
Comment (nwalberts) [inline client/src/components/NewShoeForm.js]: remove this console.log

File: client/src/components/NewShoeForm.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+				<label>Color
+					<input type="text" name="color" value={newShoe.color} onChange={handleInput}/>
+				</label>
+				<label>Category
+					<input type="text" name="category" value={newShoe.category} onChange={handleInput}/>
+				</label>
+				<label>Image Url
+					<input type="text" name="url" value={newShoe.url} onChange={handleInput}/>
+				</label>
+					<input type="submit" value="submit" className="button" onClick={handleSubmit}/>
```
Comment (nwalberts) [inline client/src/components/NewShoeForm.js]: this last input field should not be indented in

File: server/src/routes/api/v1/shoesRouter.js [hunk 6]
```diff
@@ -1,7 +1,11 @@
 import express from "express"
+import { ValidationError } from "objection"
 import { Shoe } from "../../../models/index.js"
 
 const shoesRouter = new express.Router()
+
+
+
```
Comment (nwalberts) [inline server/src/routes/api/v1/shoesRouter.js]: we only need one of these empty lines

**Review by nwalberts — CHANGES_REQUESTED** [review]

File: server/src/routes/api/v1/shoesRouter.js [hunk 7]
```diff
@@ -1,7 +1,11 @@
 import express from "express"
+import { ValidationError } from "objection"
 import { Shoe } from "../../../models/index.js"
 
 const shoesRouter = new express.Router()
+
```
Comment (johndowd) [inline server/src/routes/api/v1/shoesRouter.js:7]: remove empty spaces

**Review by johndowd — COMMENTED** [review]
looks good! just one comment!

File: client/src/components/App.js (hunk 1 above)
Comment (JusLafo) [reply client/src/components/App.js]: It only looks like that on github, it looks normal in VScode not sure why its doing that

**Pull request merged**